testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/KafkaSsl.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.tests.product.launcher.env.common;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.tests.product.launcher.env.Environment;
+import org.testcontainers.containers.BindMode;
+
+import javax.inject.Inject;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+public class KafkaSsl
+        implements EnvironmentExtender
+{
+    private final Kafka kafka;
+
+    @Inject
+    public KafkaSsl(Kafka kafka)
+    {
+        this.kafka = requireNonNull(kafka, "kafka is null");
+    }
+
+    @Override
+    public void extendEnvironment(Environment.Builder builder)
+    {
+        builder.configureContainer(Kafka.LOGICAL_NAME_KAFKA, container -> container.withEnv("KAFKA_ADVERTISED_LISTENERS", "SSL://kafka:9092")
+                .withEnv("KAFKA_SSL_KEYSTORE_FILENAME", "kafka.broker1.keystore.jks")
+                .withEnv("KAFKA_SSL_KEYSTORE_CREDENTIALS", "broker1_keystore_creds")
+                .withEnv("KAFKA_SSL_KEY_CREDENTIALS", "broker1_sslkey_creds")
+                .withEnv("KAFKA_SSL_TRUSTSTORE_FILENAME", "kafka.broker1.truststore.jks")
+                .withEnv("KAFKA_SSL_TRUSTSTORE_CREDENTIALS", "broker1_truststore_creds")
+                .withEnv("KAFKA_SSL_ENDPOINT_IDENTIFICATION_ALGORITHM", " ")
```

Comment:
why this is empty?

Reply:
Hi @kokosing , do you mean why it is set to `" "` and not `""`?

Reply:
If yes, I set it to empty `""`.

Reply:
Do you need to define this at all?

Reply:
Hi @kokosing , yes it needs to be defined explicitly since the default is set now set to HTTPS since [2.0.0](https://kafka.apache.org/documentation/#upgrade_200_notable)

Reply:
What is wrong with `https`? What does the empty value mean?

https://docs.confluent.io/platform/current/schema-registry/installation/config.html#kafkastore-ssl-endpoint-identification-algorithm 
I think we want to use https here. We want to have product tests environments to be as closest as possible to production like environments.

Reply:
https://kafka.apache.org/documentation/#security_ssl_ca

> Due to a bug in OpenSSL, the x509 module will not copy requested extension fields from CSRs into the final certificate. Since we want the SAN extension to be present in our certificate to enable hostname verification, we'll use the ca module instead. This requires some additional configuration to be in place before we generate our CA keypair.
Save the following listing into a file called openssl-ca.cnf and adjust the values for validity and common attributes as necessary. 

It will take a bit of time before I can set this up properly.

Reply:
Seems like it is fixed in OpenSSL 3.0 Alpha https://github.com/openssl/openssl/commit/b9fbacaa7bdce2083186211b3cdf8511ad4cb91d .